Objective-C/CBLArray.h

```diff
@@ -0,0 +1,82 @@
+//
+//  CBLArray.h
+//  CouchbaseLite
+//
+//  Created by Pasin Suriyentrakorn on 4/12/17.
+//  Copyright © 2017 Couchbase. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "CBLReadOnlyArray.h"
+#import "CBLArrayFragment.h"
+@class CBLSubdocument;
+@class CBLArray;
+
+NS_ASSUME_NONNULL_BEGIN
+
+/** CBLDictionary protocol defines a set of methods for getting and setting array data. */
+@protocol CBLArray <CBLReadOnlyArray, CBLArrayFragment>
+
+/** Sets an object at the given index. Setting a nil value is eqivalent to setting an NSNull object.
+    @param object   the object.
+    @param index    the index. This value must not exceed the bounds of the array. */
+- (void) setObject: (nullable id)object atIndex: (NSUInteger)index;
+
+/** Adds an object to the end of the array. Adding a nil value is equivalent
+    to adding an NSNull object. 
+    @param object   the object. */
+- (void) addObject: (nullable id)object;
+
+/** Inserts an object at the given index. Inserting a nil value is equivalent to inserting 
+    an NSNull object. 
+    @param object   the object.
+    @param index    the index. This value must not exceed the bounds of the array. */
```

Comment:
It will throw a runtime exception like IndexOutOfBoundException. I don't remember the exact name.